cdap-watchdog/src/main/java/io/cdap/cdap/logging/appender/system/CDAPLogAppender.java

```diff
@@ -151,9 +153,22 @@ public void start() {
           Executors.newSingleThreadScheduledExecutor(Threads.createDaemonThreadFactory("log-clean-up"));
         FileMetadataCleaner fileMetadataCleaner = new FileMetadataCleaner(context.getTransactionRunner());
         LogCleaner logCleaner = new LogCleaner(fileMetadataCleaner, context.getLocationFactory(),
+                                               logFileManager.getLogsDirectoryLocation(),
                                                TimeUnit.DAYS.toMillis(fileRetentionDurationDays),
+                                               FOLDER_CLEANUP_BATCH_SIZE,
                                                fileCleanupBatchSize);
-        scheduledExecutorService.scheduleAtFixedRate(logCleaner, 10, logCleanupIntervalMins, TimeUnit.MINUTES);
+        scheduledExecutorService.schedule(new Runnable() {
+          @Override
+          public void run() {
+            long delayInMillis = logCleaner.run();
```

Comment:
Use `try..finally` here. There are tons of other exceptions but IOException and we don't want it to stop cleaning if something happens

Reply:
Otherwise LGTM

Reply:
Done